src/Symfony/Component/Routing/Router.php

```diff
@@ -56,14 +50,16 @@ class Router implements RouterInterface, RequestMatcherInterface
 
     private static ?array $cache = [];
 
-    public function __construct(LoaderInterface $loader, mixed $resource, array $options = [], ?RequestContext $context = null, ?LoggerInterface $logger = null, ?string $defaultLocale = null)
-    {
-        $this->loader = $loader;
-        $this->resource = $resource;
-        $this->logger = $logger;
+    public function __construct(
+        protected LoaderInterface $loader,
+        protected mixed $resource,
+        protected array $options = [],
```

Comment:
this should not be a promoted property as the constructor was not assigning it directly.